ush/gefs_atmos_extractvars.sh

```diff
@@ -0,0 +1,102 @@
+#! /usr/bin/env bash
+################################################################################
+#   Script:    
+#
+source "${USHgfs}/preamble.sh"
+source "${USHgfs}/extractvars_tools.sh"
+fcnt=1
+dcnt=1 
+subdata=${1}
+
+cd "${subdata}" || true
+
+for outtype in "f2d" "f3d"; do
+
+  if [[ "${outtype}" == "f2d" ]];then
+    varlist=${varlist_2d}
+  elif [[ "${outtype}" == "f3d" ]];then 
+    varlist=${varlist_3d}
+    varlist_d=${varlist_3d_d}
+  fi
+
+  outdirpre="${subdata}/${outtype}"
+  if [[ ! -d "${outdirpre}" ]]; then mkdir -p "${outdirpre}"; fi 
+
+  nh=${FHMIN}
+  while [[ ${nh} -le ${FHMAX} ]];do
+    fnh=$(printf "%3.3d" "${nh}")
+
+    if [[ "${outtype}" == "f2d" ]];then
+      if [[ ${nh} -le ${FHMAXHF} ]];then
+        outres="0p25"
+      else
+        outres="0p50"
+      fi
+    elif [[ "${outtype}" == "f3d" ]];then
+      outres="1p00"
+    fi
+
+    if [[ ${nh} -lt ${FHMAXHF} ]];then
+      outfreq=${FHOUTHF}
+    else
+      outfreq=${FHOUTLF}
+    fi                                      
+
+    if [[ "${outres}" == "0p25" ]];then
+      infile1=${COMIN_ATMOS_GRIB_0p25}/gefs.${cycle}.pgrb2.${outres}.f${fnh}
+      infile2=${COMIN_ATMOS_GRIB_0p25}/gefs.${cycle}.pgrb2b.${outres}.f${fnh}
+    elif [[ "${outres}" == "0p50" ]];then
+      infile1=${COMIN_ATMOS_GRIB_0p50}/gefs.${cycle}.pgrb2.${outres}.f${fnh}
+      infile2=${COMIN_ATMOS_GRIB_0p50}/gefs.${cycle}.pgrb2b.${outres}.f${fnh}
+    elif [[ "${outres}" == "1p00" ]];then
+      infile1=${COMIN_ATMOS_GRIB_1p00}/gefs.${cycle}.pgrb2.${outres}.f${fnh}
+      infile2=${COMIN_ATMOS_GRIB_1p00}/gefs.${cycle}.pgrb2b.${outres}.f${fnh}
+    fi
```

Comment:
```suggestion
    prod_dir="COMIN_ATMOS_GRIB_${outres}"
    infile1=${!prod_dir}/${RUN}.${cycle}.pgrb2.${outres}.f${fnh}
    infile2=${!prod_dir}/${RUN}.${cycle}.pgrb2b.${outres}.f${fnh}
```
This could be compressed.

Reply:
Thanks for this suggestion. This has been compressed.